apps/web/src/services/cost-explorer/budget/budget-main/modules/budget-list/BudgetList.vue

```diff
@@ -36,40 +40,72 @@ const props = withDefaults(defineProps<Props>(), {
 const emit = defineEmits<{(e: 'update:filters', filters:ConsoleFilter[]): void;
 }>();
 
-
 const budgetUsageApiQueryHelper = new ApiQueryHelper();
-
 const state = reactive({
-    budgetUsages: [] as BudgetUsageModel[],
+    budgetUsages: [] as BudgetUsageAnalyzeModel[],
     loading: false,
     // query
-    range: {} as BudgetUsageRange,
     pageStart: 1,
     pageLimit: 24,
-    totalCount: 0,
+    more: false,
     queryStoreFilters: props.filters as ConsoleFilter[],
     period: {} as Period,
     // api request params
     sort: {
-        key: 'usage',
+        key: 'budget_usage',
         desc: true,
-    },
+    } as Query['sort'],
     budgetUsageParam: computed<BudgetUsageAnalyzeRequestParam>(() => {
-        const param = {
-            group_by: ['budget_id'],
-            include_budget_info: true,
-            ...budgetUsageApiQueryHelper.setFilters(state.queryStoreFilters)
-                .setPage(state.pageStart, state.pageLimit)
-                .setSort(state.sort.key, state.sort.desc)
-                .data,
-        } as BudgetUsageAnalyzeRequestParam;
-
-        if (state.period.start) param.start = dayjs.utc(state.period.start).format('YYYY-MM');
-        if (state.period.end) param.end = dayjs.utc(state.period.end).format('YYYY-MM');
-
-        if (state.range.min || state.range.max) param.usage_range = state.range;
+        budgetUsageApiQueryHelper
+            .setFilters(state.queryStoreFilters)
+            .setPage(state.pageStart, state.pageLimit);
+
+        const _selectMap: Record<string, string> = {};
+        BUDGET_USAGE_ANALYZE_GROUP_BY_LIST.forEach((groupBy) => {
+            _selectMap[groupBy] = groupBy;
+        });
```

Comment:
suggestion: `_selectMap` is calculated based on `BUDGET_USAGE_ANALYZE_GROUP_BY_LIST` and `BUDGET_USAGE_ANALYZE_GROUP_BY_LIST` is `const`.
and there is no mutation of `_selectMap`.
So I think it would be better if `_selectMap` is a `const`.

Reply:
thanks!!!!!!